securejarhandler/src/main/java/cpw/mods/jarhandling/impl/CompositeJarContents.java

```diff
@@ -0,0 +1,289 @@
+package cpw.mods.jarhandling.impl;
+
+import cpw.mods.jarhandling.JarContents;
+import cpw.mods.jarhandling.JarResource;
+import cpw.mods.jarhandling.JarResourceVisitor;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.file.Path;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.HexFormat;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Nullable;
+
+@ApiStatus.Internal
+public final class CompositeJarContents implements JarContents {
+    private final JarContents[] delegates;
+    @Nullable
+    private final PathFilter @Nullable [] filters;
+    private final List<Path> contentRoots;
+
+    private volatile Optional<String> checksum;
+
+    public CompositeJarContents(List<JarContents> delegates) {
+        this(delegates, null);
+    }
+
+    public CompositeJarContents(List<JarContents> delegates, @Nullable List<@Nullable PathFilter> filters) {
```

Comment:
To me it does not make sense that both the `List` and its contents are Nullable.
Please clarify in the JDOC why this is needed on this method.

Reply:
Ah okey, this is really not directly obvious.... Some more documentation would have been nice.....